### PR TITLE
[ACR] 'az acr create/update': branch removed and update task improved

### PR DESCRIPTION
**Pull request by rosanch**
**History Notes:**  

[ACR] BREAKING CHANGE: 'az acr task create/update' Branch parameter is removed
[ACR] 'az acr task update' Customer now can update context, git-token, and or triggers individually. 

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
add to S167

**Comment by northtyphoon** [comment]
@rosanch can you update the change log? other looks good to me. @shahzzzam @jaysterp

**Comment by rosanch** [comment]
> @rosanch can you update the change log? other looks good to me. @shahzzzam @jaysterp

@northtyphoon What do you mean by change log? the History.rst file? If so that file is changed automatically now. No need to update it.
Also any idea why might  be the error in the testing i ran all the exact same commands and none of them failed. I'm not sure what the problem might be.

**Comment by shahzzzam** [comment]
> > @rosanch can you update the change log? other looks good to me. @shahzzzam @jaysterp
> 
> 
> 
> @northtyphoon What do you mean by change log? the History.rst file? If so that file is changed automatically now. No need to update it.
> 
> Also any idea why might  be the error in the testing i ran all the exact same commands and none of them failed. I'm not sure what the problem might be.

I think he means the description of the Pull Request.

**Comment by qwordy** [comment]
I remember there is a similar PR?

File: src/azure-cli/azure/cli/command_modules/acr/_params.py [hunk 1]
```diff
@@ -200,7 +200,6 @@ def load_arguments(self, _):  # pylint: disable=too-many-statements
         c.argument('pull_request_trigger_enabled', arg_group='Trigger', help="Indicates whether the source control pull request trigger is enabled. The trigger is disabled by default.", arg_type=get_three_state_flag())
         c.argument('schedule', arg_group='Trigger', help="Schedule for a timer trigger represented as a cron expression. An optional trigger name can be specified using `--schedule name:schedule` format. Multiples supported by passing --schedule multiple times.", action='append')
         c.argument('git_access_token', arg_group='Trigger', help="The access token used to access the source control provider.")
-        c.argument('branch', arg_group='Trigger', deprecate_info=c.deprecate(hide=True), help="The source control branch name. Please specify your source branch in the context parameter e.g. https://github.com/Azure-Samples/acr-build-helloworld-node.git#mybranch")
```
Comment (yungezz) [inline src/azure-cli/azure/cli/command_modules/acr/_params.py]: what's behavior for user input branch already?

**Comment by yungezz** [comment]
> I remember there is a similar PR?

hi @qwordy what's the PR? could you pls share it?

File: src/azure-cli/azure/cli/command_modules/acr/_params.py (hunk 1 above)
Comment (rosanch) [reply src/azure-cli/azure/cli/command_modules/acr/_params.py]: user can set the branch of their git project where they want to read the file from.
But now we require the client to pass the branch with the context using the following format:
URI#branch:folder
For more formating details you can refer to https://docs.docker.com/engine/reference/commandline/build/#git-repositories

**Comment by rosanch** [comment]
@qwordy  do you mean the one where we deprecated the parameter?
https://github.com/Azure/azure-cli/pull/11136

**Comment by qwordy** [comment]
> 
> 
> @qwordy do you mean the one where we deprecated the parameter?
> #11136

Yes, it is. Thank you!

**Comment by qwordy** [comment]
test_acr_task fails, please fix it

**Comment by rosanch** [comment]
@qwordy Fixed. PR it's ready to merge
